torch/_jit_internal.py

```diff
@@ -627,6 +627,13 @@ def _get_overloaded_methods(method, mod_class):
 
 
 def is_tuple(ann):
+    if ann is Tuple:
```

Comment:
Makes sense. I can do this for `Dict` also.

Reply:
It's true that the error msg is better now - in the future we could add a why_not parameter return value to improve but that should probably be its own PR...

Reply:
Sorry I deleted my previous comment because I didn't understand what you were saying at first. It is possible to return `None` instead of erroring here, but it will require more code due to small differences in the `typing` module between python3.6 and python3.8. You can see a concrete example of doing this [in one of my previous diffs for `Dict`](https://github.com/pytorch/pytorch/pull/44334/commits/fa5bf34c7425e50542c36f7df95bf88999b836dd). The red diff is what I would have to add back. Take a look and if that extra code is acceptable, I can do it for `Tuple` and `List` too.